tensorflow/compiler/aot/codegen.cc

```diff
@@ -229,8 +230,9 @@ Status GenResultMethods(const tf2xla::Config& config,
   int readonly_variables = absl::c_count_if(
       config.variable(),
       [](const tf2xla::Variable& var) { return var.readonly(); });
+  const int num_results_int = num_results;
   if (config.fetch_size() + config.variable_size() - readonly_variables !=
-      num_results) {
+      num_results_int) {
```

Comment:
Better extract the condition to a separate expression above. Readability is improved from an equality checking broken into 2 lines to a computation on a line and a comparison on the next one.